### PR TITLE
Include trigonometric time features

### PR DESCRIPTION
**Pull request by AUdaltsova**
# Pull Request

## Description

Adding calculation and support for trigonometric time features (sine and cosine of time of year and time of day, 4 features total) for windnet. Currently calculated only from windnet based on wind's time index

NB: it replaces add_sun_position at `MergeNumpyModalities` in `windnet.py`; ultimately decided it's never used for wind, so not a loss. 

Added tests to check time transform works as expected

## How Has This Been Tested?

Ran training of windnet with time features included

- [X] Yes

## Checklist:

- [X] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: ocf_datapipes/transform/numpy_batch/datetime_features.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+"""Datapipes to add Sun position to NumpyBatch"""
+
+import numpy as np
+from torch.utils.data import IterDataPipe, functional_datapipe
+
+from ocf_datapipes.batch import BatchKey
+
+
+def _get_date_time_in_pi(dt):
```
Comment (Sukh-P) [inline ocf_datapipes/transform/numpy_batch/datetime_features.py:9]: Could we perhaps add a small unit test for this?

Comment (Sukh-P) [inline ocf_datapipes/transform/numpy_batch/datetime_features.py:9]: Also would be nice to have type hints on this function

Comment (AUdaltsova) [reply ocf_datapipes/transform/numpy_batch/datetime_features.py:9]: Yep good idea

File: ocf_datapipes/transform/numpy_batch/datetime_features.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            time_utc = np_batch[BatchKey.wind_time_utc]
+
+            # Check if the input is batched
+            # time_utc could have shape (batch_size, n_times) or (n_times,)
+            assert len(time_utc.shape) in [1, 2]
+            is_batched = len(time_utc.shape) == 2
+
+            times = time_utc.astype("datetime64[s]")
+
+            if is_batched:
```
Comment (Sukh-P) [inline ocf_datapipes/transform/numpy_batch/datetime_features.py:49]: Do we expect that the times will be an a numpy array here? If so would the broadcasting of the `_get_date_time_in_pi` just work for the 2 dimensional case as well? I _think_ it might so you may not need this special logic for the is batched case? Let me know if I am mistaken or I have missed something!

**Comment by peterdudfield** [comment]
does the PR description need an update?

**Review by Sukh-P — APPROVED** [review]
Great work adding this in!

File: tests/transform/numpy_batch/test_datetime_features.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import numpy as np
+from datetime import datetime
+
+from ocf_datapipes.transform.numpy_batch import AddTrigonometricDateTime
+
+from ocf_datapipes.transform.numpy_batch.datetime_features import _get_date_time_in_pi
+
+
+def test_get_date_time_in_pi():
```
[the remark this answers is not in the document]
Comment (AUdaltsova) [reply tests/transform/numpy_batch/test_datetime_features.py:6]: Yeah should've just done a 2D right away tbh

**Comment by AUdaltsova** [comment]
Ok I think this is good to go, if @peterdudfield @Sukh-P you're both happy I'll merge?
